Formula/qt5.rb

```diff
@@ -199,6 +199,13 @@ def install
               /\n# pkgconfig\n(PKG_CONFIG_(SYSROOT_DIR|LIBDIR) = .*\n){2}\n/,
               "\n"
 
+    # Make `HOMEBREW_PREFIX/lib/qt5/plugins` an additional plug-in search path
+    # for Qt Designer to support formulae that provide Qt Designer plug-ins.
+    system "/usr/libexec/PlistBuddy",
+           "-c", "Add :LSEnvironment:QT_PLUGIN_PATH string \"#{HOMEBREW_PREFIX}/lib/qt5/plugins\"",
+           "#{bin}/Designer.app/Contents/Info.plist"
+    touch "#{bin}/Designer.app" # re-register edited bundle with Launch Services
```

Comment:
Found the issue with `qt.conf`. It is read in via [QSettings](http://doc.qt.io/qt-5/qsettings.html), so Plugins subgroup in `qt.conf` _should_ be able to support QStringList and QSetting's specialized arrays. However, [QLibraryInfo](http://doc.qt.io/qt-5/qlibraryinfo.html) is set up to _always_ return a string representing a single path for each subgroup of `[PATHS]` (i.e. [QLibraryInfo::LibraryLocation](http://doc.qt.io/qt-5/qlibraryinfo.html#LibraryLocation-enum) enums). Any attempt to load a list-styled .ini value, results in a single string (not a list), which causes Designer.app to not find any plugins, thereby crashing when it can not find the required `cocoa` platform plugin.

So, using `qt.conf` is no good for setting multiple paths for plugin discovery. The `QT_PLUGIN_PATH` is always appended to configured paths, so works as needed for setting the extra Homebrew-specific path. Editing Info.plist is still seems the simplest means of passing that env var into the app bundle launch.